### ISSUE TITLE
limiting gpu usage

### ISSUE THREAD

**Issue by pypypypypypypypypyp** [lightvector/KataGo#428]
when I use the engine it makes my computer unusable while running, even when I set numSearchThreads to 1. leela has the option to change the MAX_CPUS during compilation, is there a similar option for katago?

**Comment by lightvector** [comment]
If the issue is CPU usage (rather than, for example, GPU usage causing your video display to freeze up), then with numSearchThreads = 1, I would be surprised if any sort of MAX_CPUS limitation would even help, even if it were to be implemented, because even though KataGo will have some internal helper threads that it switches between, computationally only one of those will be running at a time anywhere close to 100%.

When you visualize the usage in your system monitor, is it using more than one CPU a lot? Please double-check and make sure it's using the config you set to 1 thread - there is a chance you are running it in a way that it is not using the config that you edited, but rather using a different config set by default to have more threads.

The next thing to ask is also - are you sure it is CPU? Often the cause of a computer becoming slow and unusable is due to running out of RAM (and then everything gets stuck paging memory between RAM and disk, which is horribly slow), rather than CPU. If it is RAM, you can try smaller networks and also set `nnCacheSizePowerOfTwo` and `nnMutexPoolSizePowerOfTwo` to smaller numbers, like 16 and 12 instead of the default 20 and 16.

Lastly, if it really is CPU usage, then if you are on windows, try setting the process to low-priority: https://winaero.com/change-process-priority-windows-10/. Similarly, on Linux, I believe the "nice" command will do this. Or else you can google for whatever is relevant to your OS otherwise.

**Comment by pypypypypypypypypyp** [comment]
yes the issue is gpu usage, I thought by limiting it to one cpu thread I could limit the amount of jobs sent to the gpu, which worked with leela but not for kata, I guess kata is more efficient cpu-side, or less efficient gpu-side.

**Comment by lightvector** [comment]
Have you tried the Eigen version? That's a version that doesn't use the GPU at all, it's pure CPU. And, if your GPU is very weak, it might not even be much slower to only use CPU.
